pkg/process/procutil/process_linux.go

```diff
@@ -89,31 +92,57 @@ type Probe struct {
 // NewProcessProbe initializes a new Probe object
 func NewProcessProbe(options ...Option) *Probe {
 	hostProc := util.HostProc()
-	bootTime, _ := bootTime(hostProc)
+	bootTime, err := bootTime(hostProc)
+	if err != nil {
+		log.Errorf("could not parse boot time: %s", err)
+	}
 
 	p := &Probe{
 		procRootLoc: hostProc,
 		uid:         uint32(os.Getuid()),
 		euid:        uint32(os.Geteuid()),
-		bootTime:    bootTime,
 		clockTicks:  getClockTicks(),
+		exit:        make(chan struct{}),
 	}
+	atomic.StoreUint64(&p.bootTime, bootTime)
 
 	for _, o := range options {
 		o(p)
 	}
 
+	go p.syncBootTime()
+
 	return p
 }
 
 // Close cleans up everything related to Probe object
 func (p *Probe) Close() {
+	close(p.exit)
 	if p.procRootFile != nil {
 		p.procRootFile.Close()
 		p.procRootFile = nil
 	}
 }
 
+// syncBootTime checks bootTime every minute and stores it.
+// Make sure we get the correct boot time if the clock of the host is temporarily drifted but gets corrected later on
+func (p *Probe) syncBootTime() {
+	ticker := time.NewTicker(bootTimeRefreshInterval)
+	defer ticker.Stop()
+
+	select {
+	case <-ticker.C:
+		bootTime, err := bootTime(p.procRootLoc)
```

Comment:
I'm not sure if it's a good idea but is there a way to update the `bootTime` only when it changes? Could we assume that it has possibly changed only when the `/proc/stat` file has been modified?

If reading the `/stat` file every minute is not a concern in terms of performance, I think that it's ok to keep it as it is.